File: Code/LCD.cpp
#include <Arduino.h>
#include "LCD.h"

#include <Wire.h> 
#include <LiquidCrystal_I2C.h>

LiquidCrystal_I2C lcd(0x27,16,2);  // set the LCD address to 0x27 for a 16 chars and 2 line display

//Initializing LCD and printing welcome string
void LCD_Init(void)
{
  lcd.init();
  lcd.backlight();
  lcd.setCursor(3,0);
  lcd.print("Welcome to");
  lcd.setCursor(4,1);
  lcd.print("Extruder");
  delay(WELCOME_DELAY);
  lcd.clear();
  lcd.print("Temp:");
  lcd.setCursor(0,1);
  lcd.print("RPM:");
}
//Function to display string at exact position
void LCD_DisplayExactPosition(const String string,int x,int y)
{
  lcd.setCursor(x,y);
  lcd.print(string);
}

//Function to display string
void LCD_Display(const String string)
{
  lcd.print(string);
}

File: PythonCode/LCD.py
#importing system modules
from machine import I2C, Pin
import time
#importing i2c_lcd library
from i2c_lcd import I2cLcd 

#Default LCD Address
DEFAULT_I2C_ADDR = 0x27

#Initializing I2C
i2c = I2C(scl=Pin(5), sda=Pin(4), freq=400000) 
lcd = I2cLcd(i2c, DEFAULT_I2C_ADDR, 2, 16)

#Function to display string on LCD
def Disp(string):
    lcd.putstr(string)

#Function to display string on LCD with exact position    
def DispPos(string,x,y):
    lcd.move_to(x, y)
    lcd.putstr(string)

#Function to clear LCD	
def Clear():
    lcd.clear()

#Function to Display at the beginning of code
def Welcome(delay_time):
    DispPos("Welcome to",3,0)
    DispPos("Extruder",4,1)
    time.sleep(delay_time)
    Clear()
    Disp("Temp:")
    DispPos("RPM:",0,1)

#Function to Display required variables
def DispVars(act_temp,temp,rpm):
    DispPos(str(act_temp),5,0)
    Disp("/")
    Disp(str(temp))
    Disp(" ")
    DispPos(str(rpm),4,1)
    Disp(" ")